munge/shared/dictionary.ts

```diff
@@ -48,50 +50,100 @@ type ExportableWordform<L> = Required<Omit<Wordform<L>, "formOf">> & {
   formOf: string;
 };
 
+type NonFunctionMembers<T> = {
+  [key in keyof T]: T[key] extends Function ? never : T[key];
+};
+
+/**
+ * A non-class version of DictionaryEntry, to represent the JSON-serialized form.
+ */
+interface ExportableDictionaryEntry<L>
+  extends NonFunctionMembers<DictionaryEntry<L>> {}
+
+export type ImportJsonJsonEntry<L> =
+  | ExportableDictionaryEntry<L>
+  | ExportableWordform<L>;
+
 export class Dictionary<L = DefaultLinguistInfo> {
   /**
    * FST tags which distinguish the lexeme, e.g., +N and +V, as opposed to tags
    * that distinguish the wordform within a lexeme, e.g., +Sg and +Pl.
    */
-  readonly lexicalTags: Set<string>;
+  readonly _lexicalTags: Set<string> | null;
   _entries: (DictionaryEntry<L> | Wordform<L>)[];
-  _byText: Map<string, DictionaryEntry<L>>;
-
-  constructor(lexicalTags: string[]) {
-    this.lexicalTags = new Set(lexicalTags);
+  /* WARNING: only holds DictionaryEntry object, not Wordforms */
+  _byText: Map<string, DictionaryEntry<L>[]>;
+  _bySlug: Map<string, DictionaryEntry<L>>;
+
+  constructor(lexicalTags?: string[]) {
+    if (lexicalTags) {
+      this._lexicalTags = new Set(lexicalTags);
+    } else {
+      this._lexicalTags = null;
+    }
     this._entries = [];
     this._byText = new Map();
+    this._bySlug = new Map();
+    this._bySlug = new Map();
```

Comment:
Why is this defined twice?

Reply:
Good catch! This is an artifact of resolving some slightly hairy merge conflicts across different branches where I was working on this. I will remove it.

```suggestion
```